app/assets/javascripts/redux/modules/models/listenSignals.js

```diff
@@ -101,21 +120,37 @@ export const reducer = handleActions({
 
   [LISTEN_SIGNAL_SHOW_REQUEST_SUCCESS]: handlesListenSignalSucccesResponse,
 
-  [LISTEN_SIGNAL_SHOW_REQUEST_FAIL]: (state, action) => ({
-    ...state,
-    data: {
-      ..._.get(state, 'data', {}),
-      [action.meta.id]: {
-        ..._.get(state, `data.${action.meta.id}`, {}),
-        error: action.payload,
-        loading: false,
-        loaded: false,
-      },
-    },
-  }),
+  [LISTEN_SIGNAL_SHOW_REQUEST_FAIL]: handlesListenSignalFailResponse,
+
+  [LISTEN_SIGNALS_POST_REQUEST]: handleSignalRequest,
 
   [LISTEN_SIGNALS_POST_REQUEST_SUCCESS]: handlesListenSignalSucccesResponse,
 
+  [LISTEN_SIGNALS_POST_REQUEST_FAIL]: handlesListenSignalFailResponse,
+
+  [LISTEN_SIGNALS_PUT_REQUEST]: handleSignalRequest,
```

Comment:
This is an update so we should be modifying the `loading` state of the actual signal. The returned state should look like:

```jsx
{
  ...state,
  data: {
    ..._.get(state, 'data', {}),
    [id]: {
      ..._.get(state, `data.${action.meta.id}`, {}),
      loading: true, // updating: true ...?
      loaded: true,
    },
  },
}
```

This indicates that we've already `loaded` the signal and that we are currently loading new data on it.

One thing that we could consider is maybe adding an `updating` flag. Might be overkill but my thought is that we're not really `loading` the signal anymore as it is already loaded. Food for thought.